Async publishing from non-Windows platforms always fails during artifact creation

I attempted to on-board AspNetCore to publishing with Arcade, but immediately ran into this error message on all non-Windows platforms.

![image](https://user-images.githubusercontent.com/2696087/59453041-e3b5ea80-8dc3-11e9-81ab-186201a1c4bb.png)
https://dev.azure.com/dnceng/internal/_build/results?buildId=223311&view=logs
```
##[error]Unable to process command '  ##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]/Users/vsts/agent/2.152.1/work/1/s/eng/common\PublishToPackageFeed.proj' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)
##[error]Path not exist: /Users/vsts/agent/2.152.1/work/1/s/eng/common\PublishToPackageFeed.proj
##[error]Unable to process command '  ##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]/Users/vsts/agent/2.152.1/work/1/s/eng/common\PublishToSymbolServers.proj' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)
##[error]Path not exist: /Users/vsts/agent/2.152.1/work/1/s/eng/common\PublishToSymbolServers.proj
##[error]Unable to process command '  ##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]/Users/vsts/agent/2.152.1/work/1/s/eng/common\CheckSymbols.ps1' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)
##[error]Path not exist: /Users/vsts/agent/2.152.1/work/1/s/eng/common\CheckSymbols.ps1
```

My guess: this would be fixed by normalizing file paths to use forward slashes on Linux/macOS.


cc @JunTaoLuo @markwilkie

--- a/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj
+++ b/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj
@@ -164,7 +164,7 @@
       AssetsTemporaryDirectory="$(TempWorkingDirectory)" />
 
     <Message
-      Text="##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]$(RepositoryEngineeringDir)common\PublishToPackageFeed.proj"
+      Text="##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]$(RepositoryEngineeringDir)common/PublishToPackageFeed.proj"
       Importance="high" />
 
     <!--
@@ -181,7 +181,7 @@
       DestinationFolder="$(TempWorkingDirectory)\$(AssetManifestFileName)" />
     
     <Message
-      Text="##vso[artifact.upload containerfolder=AssetManifests;artifactname=AssetManifests]$(TempWorkingDirectory)\$(AssetManifestFileName)"
+      Text="##vso[artifact.upload containerfolder=AssetManifests;artifactname=AssetManifests]$(TempWorkingDirectory)/$(AssetManifestFileName)"
       Importance="high" />
 
     <!--
@@ -214,11 +214,11 @@
       Condition="'@(FilesToPublishToSymbolServer)' != ''"/>
 
     <Message
-      Text="##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]$(RepositoryEngineeringDir)common\PublishToSymbolServers.proj"
+      Text="##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]$(RepositoryEngineeringDir)common/PublishToSymbolServers.proj"
       Importance="high" />
 
     <Message
-      Text="##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]$(RepositoryEngineeringDir)common\CheckSymbols.ps1"
+      Text="##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]$(RepositoryEngineeringDir)common/CheckSymbols.ps1"
       Importance="high" />
 
   </Target>

--- a/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj
+++ b/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj
@@ -26,9 +26,9 @@
     <PublishToSymbolServer Condition="'$(UsingToolSymbolUploader)' == 'true' and '$(AzureFeedUrl)' == '' and '$(ContinuousIntegrationBuild)' == 'true'">true</PublishToSymbolServer>
 
     <AssetManifestFileName>$(OS)-$(PlatformName).xml</AssetManifestFileName>
-    <AssetManifestFilePath>$(ArtifactsLogDir)AssetManifest\$(AssetManifestFileName)</AssetManifestFilePath>
+    <AssetManifestFilePath>$(ArtifactsLogDir)AssetManifest/$(AssetManifestFileName)</AssetManifestFilePath>
 
-    <SymbolPackagesDir>$(ArtifactsTmpDir)SymbolPackages\</SymbolPackagesDir>
+    <SymbolPackagesDir>$(ArtifactsTmpDir)SymbolPackages/</SymbolPackagesDir>
 
     <PublishToBlobFeedFlatContainer Condition="'$(PublishToBlobFeedFlatContainer)' == ''">false</PublishToBlobFeedFlatContainer>
 
@@ -40,8 +40,8 @@
     <PublishDependsOnTargets>BeforePublish;$(PublishDependsOnTargets)</PublishDependsOnTargets>
   </PropertyGroup>
 
-  <Import Project="$(NuGetPackageRoot)microsoft.dotnet.build.tasks.feed\$(MicrosoftDotNetBuildTasksFeedVersion)\build\Microsoft.DotNet.Build.Tasks.Feed.targets" Condition="$(PublishToAzure)"/>
-  <Import Project="$(NuGetPackageRoot)microsoft.symboluploader.build.task\$(MicrosoftSymbolUploaderBuildTaskVersion)\build\PublishSymbols.targets" Condition="$(PublishToSymbolServer)"/>
+  <Import Project="$(NuGetPackageRoot)microsoft.dotnet.build.tasks.feed/$(MicrosoftDotNetBuildTasksFeedVersion)/build/Microsoft.DotNet.Build.Tasks.Feed.targets" Condition="$(PublishToAzure)"/>
+  <Import Project="$(NuGetPackageRoot)microsoft.symboluploader.build.task/$(MicrosoftSymbolUploaderBuildTaskVersion)/build/PublishSymbols.targets" Condition="$(PublishToSymbolServer)"/>
 
   <Target Name="Publish"
           DependsOnTargets="$(PublishDependsOnTargets)" />
@@ -122,7 +122,7 @@
 			<!-- When we have <account>.visualstudio.com -->
       <AzureDevOpsAccount Condition="$(CollectionUri.IndexOf('visualstudio.com')) >= 0">$(CollectionUri.Split('.')[0].Split('/')[2])</AzureDevOpsAccount>
 
-      <TempWorkingDirectory>$(ArtifactsDir)\..\AssetsTmpDir\$([System.Guid]::NewGuid())</TempWorkingDirectory>
+      <TempWorkingDirectory>$(ArtifactsDir)/../AssetsTmpDir/$([System.Guid]::NewGuid())</TempWorkingDirectory>
     </PropertyGroup>
 
     <!--
@@ -178,7 +178,7 @@
 
     <Copy 
       SourceFiles="$(AssetManifestFilePath)" 
-      DestinationFolder="$(TempWorkingDirectory)\$(AssetManifestFileName)" />
+      DestinationFolder="$(TempWorkingDirectory)/$(AssetManifestFileName)" />
     
     <Message
       Text="##vso[artifact.upload containerfolder=AssetManifests;artifactname=AssetManifests]$(TempWorkingDirectory)/$(AssetManifestFileName)"
@@ -205,7 +205,7 @@
         https://github.com/dotnet/core-eng/issues/3645
       -->
     <ItemGroup>
-      <FilesToPublishToSymbolServer Include="$(ArtifactsSymStoreDirectory)**\*.pdb"/>
+      <FilesToPublishToSymbolServer Include="$(ArtifactsSymStoreDirectory)**/*.pdb"/>
     </ItemGroup>
 
     <Message
@@ -241,7 +241,7 @@
         SymbolUploader doesn't support embedded PDBs yet, so let SymStore.targets do the conversion for now.
         https://github.com/dotnet/core-eng/issues/3645
       -->
-      <FilesToPublishToSymbolServer Include="$(ArtifactsSymStoreDirectory)**\*.pdb"/>
+      <FilesToPublishToSymbolServer Include="$(ArtifactsSymStoreDirectory)**/*.pdb"/>
 
       <!--
         Publish Portable PDBs contained in symbol packages.

--- a/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj
+++ b/src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj
@@ -26,9 +26,9 @@
     <PublishToSymbolServer Condition="'$(UsingToolSymbolUploader)' == 'true' and '$(AzureFeedUrl)' == '' and '$(ContinuousIntegrationBuild)' == 'true'">true</PublishToSymbolServer>
 
     <AssetManifestFileName>$(OS)-$(PlatformName).xml</AssetManifestFileName>
-    <AssetManifestFilePath>$(ArtifactsLogDir)AssetManifest/$(AssetManifestFileName)</AssetManifestFilePath>
+    <AssetManifestFilePath>$(ArtifactsLogDir)AssetManifest\$(AssetManifestFileName)</AssetManifestFilePath>
 
-    <SymbolPackagesDir>$(ArtifactsTmpDir)SymbolPackages/</SymbolPackagesDir>
+    <SymbolPackagesDir>$(ArtifactsTmpDir)SymbolPackages\</SymbolPackagesDir>
 
     <PublishToBlobFeedFlatContainer Condition="'$(PublishToBlobFeedFlatContainer)' == ''">false</PublishToBlobFeedFlatContainer>
 
@@ -40,8 +40,8 @@
     <PublishDependsOnTargets>BeforePublish;$(PublishDependsOnTargets)</PublishDependsOnTargets>
   </PropertyGroup>
 
-  <Import Project="$(NuGetPackageRoot)microsoft.dotnet.build.tasks.feed/$(MicrosoftDotNetBuildTasksFeedVersion)/build/Microsoft.DotNet.Build.Tasks.Feed.targets" Condition="$(PublishToAzure)"/>
-  <Import Project="$(NuGetPackageRoot)microsoft.symboluploader.build.task/$(MicrosoftSymbolUploaderBuildTaskVersion)/build/PublishSymbols.targets" Condition="$(PublishToSymbolServer)"/>
+  <Import Project="$(NuGetPackageRoot)microsoft.dotnet.build.tasks.feed\$(MicrosoftDotNetBuildTasksFeedVersion)\build\Microsoft.DotNet.Build.Tasks.Feed.targets" Condition="$(PublishToAzure)"/>
+  <Import Project="$(NuGetPackageRoot)microsoft.symboluploader.build.task\$(MicrosoftSymbolUploaderBuildTaskVersion)\build\PublishSymbols.targets" Condition="$(PublishToSymbolServer)"/>
 
   <Target Name="Publish"
           DependsOnTargets="$(PublishDependsOnTargets)" />
@@ -122,7 +122,7 @@
 			<!-- When we have <account>.visualstudio.com -->
       <AzureDevOpsAccount Condition="$(CollectionUri.IndexOf('visualstudio.com')) >= 0">$(CollectionUri.Split('.')[0].Split('/')[2])</AzureDevOpsAccount>
 
-      <TempWorkingDirectory>$(ArtifactsDir)/../AssetsTmpDir/$([System.Guid]::NewGuid())</TempWorkingDirectory>
+      <TempWorkingDirectory>$(ArtifactsDir)\..\AssetsTmpDir\$([System.Guid]::NewGuid())</TempWorkingDirectory>
     </PropertyGroup>
 
     <!--
@@ -178,7 +178,7 @@
 
     <Copy 
       SourceFiles="$(AssetManifestFilePath)" 
-      DestinationFolder="$(TempWorkingDirectory)/$(AssetManifestFileName)" />
+      DestinationFolder="$(TempWorkingDirectory)\$(AssetManifestFileName)" />
     
     <Message
       Text="##vso[artifact.upload containerfolder=AssetManifests;artifactname=AssetManifests]$(TempWorkingDirectory)/$(AssetManifestFileName)"
@@ -205,7 +205,7 @@
         https://github.com/dotnet/core-eng/issues/3645
       -->
     <ItemGroup>
-      <FilesToPublishToSymbolServer Include="$(ArtifactsSymStoreDirectory)**/*.pdb"/>
+      <FilesToPublishToSymbolServer Include="$(ArtifactsSymStoreDirectory)**\*.pdb"/>
     </ItemGroup>
 
     <Message
@@ -241,7 +241,7 @@
         SymbolUploader doesn't support embedded PDBs yet, so let SymStore.targets do the conversion for now.
         https://github.com/dotnet/core-eng/issues/3645
       -->
-      <FilesToPublishToSymbolServer Include="$(ArtifactsSymStoreDirectory)**/*.pdb"/>
+      <FilesToPublishToSymbolServer Include="$(ArtifactsSymStoreDirectory)**\*.pdb"/>
 
       <!--
         Publish Portable PDBs contained in symbol packages.

